add a helper script to print digests of downloaded artifacts

tuminoid opened the pull request:
For updating pinned binary downloads, getting the correct digest has been more work than it needs to be.

Workflow:
- update the version numbers in `lib/common.sh`
- run `./hack/print_checksums.sh` 
- update the sha256 in `lib/common.sh` as needed

tuminoid commented:
/test-ubuntu-e2e-integration-main 
/test-centos-integration-release-1-4

tuminoid commented:
/cc @kashifest 

Would this be good enough, or have ideas to make it even simpler?

tuminoid commented:
/hold
For opinions and improvements.

tuminoid commented:
/test-ubuntu-e2e-integration-main
/test-centos-integration-release-1-5

tuminoid commented:
/test-ubuntu-e2e-integration-main
/test-centos-integration-release-1-5

lentzi90 reviewed (commented):
/approve

The print for krew was a bit cryptic compared to the rest, but I don't think it is an issue.
```
info: sha256(/tmp/tmp.Zld3ATZ9Uz/krew-linux_amd64.tar.gz): 5df32eaa0e888a2566439c4ccb2ef3a3e6e89522f2f2126030171e2585585e4f
```

tuminoid commented:
> /approve
> 
> The print for krew was a bit cryptic compared to the rest, but I don't think it is an issue.
> 
> ```
> info: sha256(/tmp/tmp.Zld3ATZ9Uz/krew-linux_amd64.tar.gz): 5df32eaa0e888a2566439c4ccb2ef3a3e6e89522f2f2126030171e2585585e4f
> ```

True... I'll cut the leading directories off from the print.

kashifest commented:
lgtm

tuminoid commented:
OK, I changed the printout not to print the temporary path, and also it deletes the artifacts instead of leaving them in the working dir. I also added a `hack/print_checksums.sh` so you don't need to source stuff.

Unfortunately our dev-env `lib/common.sh` has idiotic passwordless sudo check, so unless you've configured passwordless sudo just the exact way it wants you to, you cannot run the script. Need to fix that in another PR.

lentzi90 commented:
> Unfortunately our dev-env lib/common.sh has idiotic passwordless sudo check, so unless you've configured passwordless sudo just the exact way it wants you to, you cannot run the script. Need to fix that in another PR.

True, I'm always fighting this... :disappointed: I don't want passwordless sudo on my laptop.

tuminoid commented:
> > Unfortunately our dev-env lib/common.sh has idiotic passwordless sudo check, so unless you've configured passwordless sudo just the exact way it wants you to, you cannot run the script. Need to fix that in another PR.
> 
> True, I'm always fighting this... 😞 I don't want passwordless sudo on my laptop.

Created an issue to fix it: https://github.com/metal3-io/metal3-dev-env/issues/1288

Rozzii commented:
/lgtm

tuminoid commented:
/unhold
